rkt/trust.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+// implements https://github.com/coreos/rocket/issues/367
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/coreos/rocket/Godeps/_workspace/src/golang.org/x/crypto/openpgp"
+	"github.com/coreos/rocket/pkg/keystore"
+)
+
+var (
+	cmdTrust = &Command{
+		Name:    cmdTrustName,
+		Summary: "Trust a key for image verification",
+		Usage:   "[--prefix PREFIX] [--do-not-ask-confirmation] PUBKEY",
+		Run:     runTrust,
+	}
+	flagPrefix string
+	flagAccept bool
+)
+
+const (
+	cmdTrustName = "trust"
+)
+
+func init() {
+	commands = append(commands, cmdTrust)
+	cmdTrust.Flags.StringVar(&flagPrefix, "prefix", "", "prefix to limit trust to")
+	cmdTrust.Flags.BoolVar(&flagAccept, "do-not-ask-confirmation", false, "suppress interactive confirmation of key addition")
+}
+
+func runTrust(args []string) (exit int) {
+	if len(args) != 1 {
+		printCommandUsageByName(cmdTrustName)
+		return 1
+	}
+
+	pkl, err := getPubKeyLocation(flagPrefix, args)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error determining key location: %v\n", err)
+		return 1
+	}
+
+	pk, err := getPubKey(pkl)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error accessing key: %v\n", err)
+		return 1
+	}
+	defer pk.Close()
+
+	accepted, err := reviewKey(pk, flagAccept)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reviewing key: %v\n", err)
+	}
+
+	if !accepted {
+		fmt.Fprintf(os.Stderr, "Not trusting %q.\n", pkl)
+		return 0
+	}
+
+	fmt.Printf("Trusting %q for prefix %q.\n", pkl, flagPrefix)
+
+	if err := addPubKey(flagPrefix, pk); err != nil {
+		fmt.Fprintf(os.Stderr, "Error adding key: %v\n", err)
+		return 1
+	}
+
+	return 0
+}
+
+// getPubKeyLocation either returns the location supplied in argv or discovers one @ prefix
+func getPubKeyLocation(prefix string, args []string) (string, error) {
+	if len(args) == 1 {
+		return args[0], nil
+	}
+
+	if len(args) > 1 {
```

Comment:
line 56 goes away when --prefix PUBKEY discovery arrives, getPubKeyLocation() is just primed for it.